shared/Cargo.toml

```diff
@@ -10,6 +10,7 @@ anyhow = "1.0"
 assert_approx_eq = "1.1"
 async-trait = "0.1"
 atty = "0.2"
+bigdecimal = "0.2"
```

Comment:
Why do we need bigdecimal here? I see that we define some conversions but where do we actually use the BigDecimal? The reason I am asking is that BigDecimal was originally only used when interfacing with postgres because that is the only big integer type that Postgres supports. I feel that in other code we should be using https://docs.rs/num-bigint/0.4.0/num_bigint/index.html big integers because that library is in a better state overall.

Reply:
We use `BigDecimal` to model weights, as they tend to have fractional components. Specifically, a 2-token Balancer pool follows the following equation:
```
(reserve0 - out0)^w0 * (reserve1 + in1)^w1 = K
```

Where
```
w0 + w1 = 1.0
```

For balancer, they are represented as `uint256` where the actual value it represents is `uint256(value) / 1e18`. That is, a `uint256` value of `8e17` represents `0.8` in the above equation.

Another option would be to use `num::Ratio` instead of `BigNumber`. I was looking into the `bigdecimal` issues and PRs and it looks abandoned by the developer.

Reply:
The only other issue, is that we need `BigDecimal` for serializing input to the solver as it expects decimal notation for the weights. But we can just do the conversion locally to that module.

Reply:
Yes that's why I didn't want to use it in other places. If you need fractional numbers then num_rational::BigRational would be better.

Reply:
BIgDecimal will be used in a follow up PR where we send WeightedPool's token weights (Bfp) to the HTTP solver via JSON. Which they expect as a decimal string.

Note that with the conversions we have internally it was not possible to use anything but `use num_bigint::BigInt as NumBigInt;`. I guess when you say other code, you mean the code that is not using `num_bigint`, in which case that is something I would be happy to fix in a separate PR.